rfcs/45-stageless.md

```diff
@@ -2,488 +2,457 @@
 
 ```rust
 #[derive(SystemLabel)]
 enum Physics {
-    Forces,
-    CollisionDetection,
-    CollisionHandling,
+    ComputeForces,
+    DetectCollisions,
+    HandleCollisions,
 }
 
-impl Plugin for PhysicsPlugin{
+impl Plugin for PhysicsPlugin {
     fn build(app: &mut App){
-        // Within each frame, physics logic needs to occur after input handling, but before rendering
-        let mut common_physics_config = SystemConfig::new().after(InputSet::ReadInputHandling).before(CoreSet::Rendering);
+        // Scheduling configs are reusable.
+        let mut shared_config = Config::new()
+            // Built-in fixed update set (replaces FixedTimestep).
+            .in_set(CoreSet::FixedUpdate)
+            .after(InputSet::ReadInputHandling);
 
         app
-        // We can reuse this shared configuration on each of our sets
-        .configure_set(Physics::Forces.configure(common_physics_config).before(Physics::CollisionDetection))
-        .configure_set(Physics::CollisionDetection.configure(common_physics_config).before(Physics::CollisionHandling))
-        .configure_set(Physics::CollisionHandling.configure(common_physics_config))
-        // And then apply that config to each of the systems that are part of this set
-        .add_system(gravity.in_set(Physics::Forces))
-        // These systems have a linear chain of ordering dependencies between them
-        // Systems earlier in the chain must run before those later in the chain
-        // Other systems can run in between these systems
-        .add_systems((broad_pass, narrow_pass).chain().in_set(Physics::CollisionDetection))
-        // Add multiple systems at once to reduce boilerplate!
-        .add_systems((compute_forces, collision_damage).in_set(Physics::CollisionHandling));
+            .add_set(Physics::ComputeForces.configure_with(shared_config).before(Physics::DetectCollisions))
+            .add_set(Physics::DetectCollisions.configure_with(shared_config).before(Physics::HandleCollisions))
+            .add_set(Physics::HandleCollisions.configure_with(shared_config))
+            .add_system(gravity.in_set(Physics::ComputeForces))
+            .add_many(chain![broad_pass, narrow_pass, solve_constraints].in_set(Physics::DetectCollisions))
+            .add_system(collision_damage.in_set(Physics::HandleCollisions));
     }
 }
 ```
 
-Just like systems, system sets can be given run criteria and be part of other sets themselves, allowing users to describe properties hierarchically.
+### Configuring run conditions (run criteria)
 
-#### System ordering
+While ordering constraints determine *when* a system runs, **run conditions** determine *if* it runs at all. Run conditions consist of systems that have read-only data access and return `bool`. A system or set can have any number of conditions, and it will only run if all of them return `true`. If any condition returns `false`, the system (or systems under the set) will be skipped.

```

Comment:
```suggestion
While ordering constraints determine *when* a system runs, **run conditions** determine *if* it runs at all. Run conditions consist of systems that have read-only data access and return `bool`.
A system or set can have any number of conditions, and it will only run if all of them return `true`. 
If any condition returns `false`, the system (or systems under the set) will be skipped.
```